src/main.rs

```diff
@@ -25,23 +30,22 @@ fn main() {
  */
 
 // Player Type
-struct LeftPlayer;
-struct RightPlayer;
-
-// Positions are percentage based
-struct Position {
-    x: f32,
-    y: f32,
+enum PlayerType {
+    LeftPlayer,
+    RightPlayer,
+}
+pub struct Player {
+    player_type: PlayerType,
 }
 
-// Vectors components are described as scalars between [0,1]
-// with two decimal places of precision
-struct Vector {
+// Positions are percentage based
+pub struct Position {
     x: f32,
     y: f32,
 }
 
-struct Ball;
+pub struct Velocity(Vec2);
```

Comment:
What are you thoughts on my adding `speed` as a separate component? The ball moves a bit fast right now!

`Velocity` is a unit vector and `[-1, 1]` and speed is a positive scalar `[0, infinity)`